Chatbot Function

natalie2nguyen opened the pull request:
- Incorporated model to match message to intent. 
- Created function to match intent to response.

natalie2nguyen committed:
updated chat bot function to incorporate model's intent prediction. Also, added match_intent_to_response function to output the correct response message. Test included as well.

natalie2nguyen committed:
Added chatbot functionality to use model to predict intent and match it to the correct response to output to the user.

added more to test; mock excel sheet data

natalie2nguyen committed:
added more to test; mock excel sheet data

Ry-ka commented on server/src/chatbot.py:
```diff
@@ -32,21 +38,20 @@ def preprocess_input(user_message):
+def match_intent_to_response(intent):
+
+    file_path = Path(__file__).parent / 'dataset/OIT Responses.xlsx'
+    df = pd.read_excel(file_path, sheet_name="Sheet1")    
```
Let's not read from and build the dictionary every time a response is requested, but rather when the chatbot is spun up.

natalie2nguyen committed:
created new function to read excel sheet with intent and responses. only requires the exel to be read once instad of each time the intent is matched to the response

natalie2nguyen committed:
created new function to read excel sheet with intent and responses. only requires the exel to be read once instad of each time the intent is matched to the response

fixed tests

fixed chatbot functionality so that the intent/response dictionary isnt being created each time the response is being predicted

fixed app file

fixed app.py file again

read_data() called before request

fixed app.py again

reverted back "finally{}"

added more styling to frontend

Moved css file to correct location

changed colors of chat bubbles

fixed failing test in test_app.py

natalie2nguyen committed:
fixed tests

Ry-ka commented on server/src/chatbot.py:
```diff
@@ -18,7 +18,8 @@ def chatbot(user_message):
     rf_model, tfidf_vect = model()
     intent = predict_intent(rf_model, tfidf_vect,preprocessed_message)
-    output = match_intent_to_response(intent)
+    itent_dictionary = read_data()
```
I didn't mean encapsulate it in a function. I mean I don't want it to read from the excel file every time a request is made. We can build the dictionary once and then search it (O(1) operation) every time a request is made.

natalie2nguyen committed:
fixed chatbot functionality so that the intent/response dictionary isnt being created each time the response is being predicted

Ry-ka commented on server/src/app.py:
```diff
@@ -1,14 +1,15 @@
 def chat():
     user_message = request.json.get('message')
-    bot_response = chatbot(user_message)
+    response_dictionary = read_data()
```
chat() gets called every time a request is made. This still creates a dictionary every time a post request is made.

natalie2nguyen committed:
fixed app file

Ry-ka commented on server/src/app.py:
```diff
@@ -1,14 +1,33 @@
 def chat():
     user_message = request.json.get('message')
-    bot_response = chatbot(user_message)
+    response_dictionary = read_data()
```
You initialized the dictionary correctly, but it still reads the excel file every time chat() is called.

natalie2nguyen committed:
fixed app.py file again

natalie2nguyen committed:
read_data() called before request

Ry-ka commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -39,10 +39,10 @@ const HelpDeskWidget: React.FC = () => {
-        setMessage('');
-      }
+      
+      // Clearing out the messages to allow for new input
```
Why did you get rid of the finally block?

natalie2nguyen replied:
because after you send that message it doesnt clear instantaneously, after i removed "finally" it did

Ry-ka replied:
Sorry, what is "it"?

natalie2nguyen replied:
when you click send, the message remains in the bar for a few seconds, after removing "finally" the message clears instantaneously. if you want me to revert it, i will.

Ry-ka replied:
I'm not seeing that on my end. Could you try again with the finally block in there?

Ry-ka replied:
We need that to enforce the clear just in case an error is thrown in the try block.

Ry-ka commented on server/src/app.py:
```diff
@@ -1,14 +1,33 @@
 def chat():
-    user_message = request.json.get('message')
-    bot_response = chatbot(user_message)
+    read_data()
```
This is devolving into something strange. 

Remove the read_data call from chat(). Call it after read_data() but before @app.route

Remove the check to see if the intent_response == none in the read_data function. It lost atomicity after you added that.

Remove the white space between @app.route and def chat(). It's a decorator that mimics annotations and are meant to be paired together.

Ry-ka commented on server/src/chatbot.py:
```diff
@@ -32,21 +38,11 @@ def preprocess_input(user_message):
+    if intent in intent_response:
+        return intent_response[intent]
+    else: 
+        return "I'm sorry, I'm not sure if I understand. Could you provide me with more details please?"
```
Say, that the conversation goes like this.

```
"Is the wifi down? I can't seem to connect"

"I'm sorry, I'm not sure if I understand. Could you provide me with more details please?"

"Well I tried reconnecting several times, but it doesn't seem to be working."
```

We don't have the functionality to recognize that something is a detail to an inquiry rather than a new inquiry. I think we need to change the "... more details please?" to "Try again" or something along those lines.

natalie2nguyen committed:
fixed app.py again

natalie2nguyen committed:
reverted back "finally{}"

Ry-ka reviewed (approved):
Make sure you rebase and squash before you merge.

natalie2nguyen committed:
Merge branch 'main' into natalie_new

natalie2nguyen committed:
fixed failing test in test_app.py
